lib/src/components/NxSidebarNavigation/NxSidebarNavigation.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import classnames from 'classnames';
+import { faBars, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+
+import NxButton from '../NxButton/NxButton';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { Props, SidebarNavLinkProps, propTypes } from './types';
+
+import './NxSidebarNavitation.scss';
+
+const NxSidebarNavLink = function NxSidebarNavLink({ name, href, icon, current }: SidebarNavLinkProps) {
+  const navigationLinkClassnames = classnames('nx-sidebar__item', {
+    'selected': current
+  });
+
+  return (
+    <a href={href} className={navigationLinkClassnames} key={`nav-link-${name}`}>
+      <NxFontAwesomeIcon icon={icon} />
+      <span className="nx-sidebar__link-name">{name}</span>
+    </a>
+  );
+};
+
+const NxSidebarNavigation = function NxSidebarNavigation(props: Props) {
+  const {
+    isOpen,
+    className,
+    onToggleClick,
+    logoImg,
+    logoText,
+    logoLink,
+    links,
+    helpLink,
+    helpText,
+    collapsedReleaseText,
+    expandedReleaseText,
+    attributions
+  } = props;
+
+  const classes = classnames(className, 'nx-sidebar', {
```

Comment:
The overall class naming here is inevitably going to be tricky, since we've already got another sidebar - `.nx-page-sidebar`.  But I've just realized, do those really need to be two separate sidebars? Or could this be a modifier of that one (e.g. `.nx-page-sidebar--navigation`).  Here's what I'm picturing:

What we have today: `.nx-page` which wraps `.nx-page-header` and `.nx-page-contents`, and `.nx-page-contents` wraps `.nx-page-sidebar` and `.nx-page-main`.

What we could have going forward, still supporting the old way as well:
`.nx-page` still contains `.nx-page-contents` but `.nx-page-header` is optional.  `.nx-page-contents` has  modifier that allows it to go full-width as opposed to the old centered, constrained width style. `.nx-page-sidebar` is still the sidebar of choice, but has a modifier and specific substructure by which it can be the sidebar being implemented in this PR.

I feel like this would maximize consistency/compatibility with what we already have, and may cut down on some of the styling needed on the sidebar element itself.

Reply:
> The overall class naming here is inevitably going to be tricky, since we've already got another sidebar - `.nx-page-sidebar`. But I've just realized, do those really need to be two separate sidebars? Or could this be a modifier of that one (e.g. `.nx-page-sidebar--navigation`). Here's what I'm picturing:
> 
> What we have today: `.nx-page` which wraps `.nx-page-header` and `.nx-page-contents`, and `.nx-page-contents` wraps `.nx-page-sidebar` and `.nx-page-main`.
> 
> What we could have going forward, still supporting the old way as well:
> `.nx-page` still contains `.nx-page-contents` but `.nx-page-header` is optional. `.nx-page-contents` has modifier that allows it to go full-width as opposed to the old centered, constrained width style. `.nx-page-sidebar` is still the sidebar of choice, but has a modifier and specific substructure by which it can be the sidebar being implemented in this PR.
> 
> I feel like this would maximize consistency/compatibility with what we already have, and may cut down on some of the styling needed on the sidebar element itself.

Is this more or less what you had in mind? b9b047801218fedc6d13234b0795b2c66b9c2238